tensorflow/python/ops/bincount_ops_test.py

```diff
@@ -163,6 +164,188 @@ def test_dense_input(self,
+                       axis=-1):
+    @def_function.function(jit_compile=False)
+    def f(x,
+          weights=weights,
+          minlength=minlength,
+          maxlength=maxlength,
+          binary_output=binary_output,
+          axis=axis):
+      y = bincount_ops.bincount(
+            x,
+            weights=weights,
+            minlength=minlength,
+            maxlength=maxlength,
+            binary_output=binary_output,
+            axis=axis
+        )
+      return y
+    y = f(x,           
+          weights=weights,
+          minlength=minlength,
+          maxlength=maxlength,
+          binary_output=binary_output,
+          axis=axis)
+    self.assertAllEqual(expected_values, y)
+  @parameterized.named_parameters(    
+  {
+      "testcase_name": "_no_maxlength_small",
+      "x": np.random.randint(100, size=(300,300), dtype=np.int32)
+  }, {
+      "testcase_name": "_no_maxlength_medium",
+      "x": np.random.randint(200, size=(600,500), dtype=np.int32)
+  }, {
+      "testcase_name": "_no_maxlength_large",
+    "x": np.random.randint(500, size=(1000,1000), dtype=np.int32)
+  })
+  @test_util.disable_mlir_bridge('TODO: ?')
+  def test_compiled_dense_perf(self,
+                       x,
+                       minlength=None,
+                       maxlength=None,
+                       binary_output=False,
+                       weights=None,
+                       axis=-1):
+
+    @def_function.function(jit_compile=True)
+    def f_compiled(x,
+          weights=weights,
+          minlength=minlength,
+          maxlength=maxlength,
+          binary_output=binary_output,
+          axis=axis):
+      y = bincount_ops.bincount(
+            x,
+            weights=weights,
+            minlength=minlength,
+            maxlength=maxlength,
+            binary_output=binary_output,
+            axis=axis
+        )
+      return y
+    
+    def f(x,
+          weights=weights,
+          minlength=minlength,
+          maxlength=maxlength,
+          binary_output=binary_output,
+          axis=axis):
+      y = bincount_ops.bincount(
+            x,
+            weights=weights,
+            minlength=minlength,
+            maxlength=maxlength,
+            binary_output=binary_output,
+            axis=axis
+        )
+      return y
+
+    lambda_f = lambda: f(x,           
+                weights=weights,
+                minlength=minlength,
+                maxlength=maxlength,
+                binary_output=binary_output,
+                axis=axis)
+
+    lambda_fc = lambda: f_compiled(x,           
+                        weights=weights,
+                        minlength=minlength,
+                        maxlength=maxlength,
+                        binary_output=binary_output,
+                        axis=axis)
+    # warm-up
+    lambda_f(); lambda_fc()
+    not_compiled = timeit.timeit(lambda_f, number=300)
+    compiled = timeit.timeit(lambda_fc, number=300)
```

Comment:
On CPU:
XLA Compiled: 0.711194 Not compiled: 1.163104
XLA Compiled: 0.293870 Not compiled: 0.432131
XLA Compiled: 0.201019 Not compiled: 0.237024